c/examples/cpp_sorting_example.cpp

```diff
@@ -0,0 +1,121 @@
+#include <cstddef>
+#include <vector>
+#include <algorithm>
+#include <stdexcept>
+#include <sstream>
+#include <iostream>
+#include <type_traits>
+#include <tskit.h>
+
+static void
+handle_tskit_return_code(int code)
+{
+    if (code != 0) {
+        std::ostringstream o;
+        o << tsk_strerror(code);
+        throw std::runtime_error(o.str());
+    }
+}
+
+struct edge_plus_time {
+    double time;
+    tsk_id_t parent, child;
+    double left, right;
+};
+
+int
+sort_edges(tsk_table_sorter_t *sorter, tsk_size_t start)
+{
+    // if (sorter->tables->edges.metadata != nullptr) {
```

Comment:
Better to say ``sorter->tables->edges.metadata_length != 0``, as this will always work.

Reply:
Ah, that makes sense.  The exception was triggered by output from msprime.